extra_data/keydata.py

```diff
@@ -155,21 +165,32 @@ def source_file_paths(self):
         from pathlib import Path
         return [Path(p) for p in paths]
 
+    def _find_attributes(self, dset):
+        """Find Karabo attributes belonging to a dataset."""
+        attrs = dict(dset.attrs)
+
+        if self.is_control and self.key.endswith('.value'):
+            # For CONTROL sources, most of the attributes are saved on
+            # the parent group rather than the .value dataset.
+            attrs.update(dict(dset.parent.attrs))
```

Comment:
Nice, I either didn't know or had forgotten that datasets had a `.parent` attribute. :laughing: